src/logic/calculate.js

```diff
@@ -12,8 +11,10 @@ const calculate = (data, btnName) => {
 
   // Add a dot at the end of the recently typed number
   if (btnName === '.') {
+    if (!next && !total) return { total, next: `${0}${btnName}`, operation };
     if (!next) return { total: `${total}${btnName}`, next, operation };
     if (!total) return { total, next: `${next}${btnName}`, operation };
+    if (operation && total && (next !== '0' || next === '0')) return { total, next: `${next}${btnName}`, operation };
   }
```

Comment:
- [ ] <b>OPTIONAL:</b> There is no need to use `if condition` and `switch case` separately in this function, you can simply move these conditions in the switch case below.